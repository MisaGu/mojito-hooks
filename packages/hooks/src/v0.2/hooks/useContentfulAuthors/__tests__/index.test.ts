import { act, renderHook } from '@testing-library/react-hooks';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useContentfulAuthors,
  UseContentfulAuthorsProps,
  UseContentfulAuthorsReturn,
} from '../useContentfulAuthors';

describe('useContentfulAuthors()', () => {
  it('should be defined', () => {
    expect(useContentfulAuthors).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseContentfulAuthorsProps?],
      UseContentfulAuthorsReturn
    >(
      (args = []) => {
        return useContentfulAuthors(...args);
      },
      {
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
