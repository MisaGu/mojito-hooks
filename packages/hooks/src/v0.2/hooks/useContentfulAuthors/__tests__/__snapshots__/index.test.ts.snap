// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContentfulAuthors() returns the right result when needed 1`] = `
Object {
  "authors": Array [
    Object {
      "about": "...",
      "avatar": Object {
        "title": "Mock Author 1",
        "url": "https://mock-cdn/mock-author-1-avatar.png",
      },
      "linkedFrom": Object {
        "lotCollection": Object {
          "items": Array [
            Object {
              "mojitoId": "mock-item-id-1",
              "slug": "mock-item-1",
              "title": "Mock Item 1",
            },
          ],
        },
      },
      "name": "Mock Author 1",
      "slug": "mock-author-1",
      "sys": Object {
        "publishedAt": "2021-11-24T11:38:28.018Z",
      },
    },
    Object {
      "about": "...",
      "avatar": Object {
        "title": "Mock Author 2",
        "url": "https://mock-cdn/mock-author-2-avatar.png",
      },
      "linkedFrom": Object {
        "lotCollection": Object {
          "items": Array [
            Object {
              "mojitoId": "mock-item-id-2",
              "slug": "mock-item-2",
              "title": "Mock Item 2",
            },
            Object {
              "mojitoId": "mock-item-id-3",
              "slug": "mock-item-3",
              "title": "Mock Item 3",
            },
          ],
        },
      },
      "name": "Mock Author 2",
      "slug": "mock-author-2",
      "sys": Object {
        "publishedAt": "2021-11-24T11:38:28.018Z",
      },
    },
  ],
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
